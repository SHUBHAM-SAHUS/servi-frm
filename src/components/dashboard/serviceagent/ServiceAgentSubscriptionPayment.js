import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { ADMIN_DETAILS } from '../../../dataProvider/constant';
import ServiceAgentMakePayment from '../serviceagent/ServiceAgentMakePayment';
import ServiceAgentSaveCard from '../serviceagent/ServiceAgentSaveCard';
import * as orgBillingActions from "../../../actions/organizationBillingAction";
import {getStorage, setStorage} from '../../../utils/common';

class ServiceAgentSubscriptionPayment extends React.Component {

    constructor(props) {
        super(props);
        this.state = { showSavedCards: false }
        this.getCardDetails()
    }

    getCardDetails = () => {
        var organId = JSON.parse(getStorage(ADMIN_DETAILS)) ? JSON.parse(getStorage(ADMIN_DETAILS)).organisation.id : null
        this.props.billingAction.getOrganizationBillingDetails(this.props.agentID).then((message) => {
            if (this.props.billingDetails.card_details != null) {
                this.setState({
                    showSavedCards: false
                })
            } else {
                this.setState({
                    showSavedCards: true
                })
            }
        }).catch((message) => {
        });
    }

    showDifferentCard = () => {
        this.setState({
            showSavedCards: true
        })
    }

    showSavedCardPage = () => {
        this.setState({
            showSavedCards: false
        })
    }

    render() {
        return (
            <div>
                {(this.props.billingDetails.card_details != null) ?
                    (this.state.showSavedCards ?
                        <ServiceAgentSaveCard agentID={this.props.agentID} subscriptionID={this.props.subscriptionID} showCard={this.showSavedCardPage} dissmissSubscriptionPayment={this.props.dissmissSubscriptionPayment} />
                        : <ServiceAgentMakePayment agentID={this.props.agentID} subscriptionID={this.props.subscriptionID} showDifferentCard={this.showDifferentCard} dissmissSubscriptionPayment={this.props.dissmissSubscriptionPayment} />
                    ) : <ServiceAgentSaveCard agentID={this.props.agentID} subscriptionID={this.props.subscriptionID} showCard={this.showSavedCardPage} dissmissSubscriptionPayment={this.props.dissmissSubscriptionPayment} />}
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        billingDetails: state.organizationBilling.billingDetails,
    }
}

const mapDispatchToprops = dispatch => {
    return {
        billingAction: bindActionCreators(orgBillingActions, dispatch)
    }
}

export default connect(mapStateToProps, mapDispatchToprops)(ServiceAgentSubscriptionPayment);;