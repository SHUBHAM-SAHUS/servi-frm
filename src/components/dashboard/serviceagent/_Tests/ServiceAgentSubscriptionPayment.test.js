import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ServiceAgentSubscriptionPayment from '../ServiceAgentSubscriptionPayment';

describe('ServiceAgent ServiceAgentSubscriptionPayment Component Test', () => {
    const initialState = {
        organizationBilling: {
            billingDetails: {},
        },
    }

    const mockStore = configureStore()
    let store, container

    beforeEach(() => {
        store = mockStore(initialState)
        container = shallow(<ServiceAgentSubscriptionPayment store={store} />)
    })

    it('should render component ServiceAgentSubscriptionPayment correctly', () => {
        const component = shallow(<ServiceAgentSubscriptionPayment store={store} />);
        expect(component).toMatchSnapshot();
    });

    // it('Render the connected(SMART) ServiceAgentSaveCard component', () => {
    //     expect(container.length).toEqual(1)
    // });

    // it('Render the connected(SMART) ServiceAgentSaveCard component', () => {
    //     expect(container.length).toEqual(1)
    // });
})

