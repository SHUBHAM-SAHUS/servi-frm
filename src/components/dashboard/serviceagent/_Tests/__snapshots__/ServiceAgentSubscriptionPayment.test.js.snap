// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ServiceAgent ServiceAgentSubscriptionPayment Component Test should render component ServiceAgentSubscriptionPayment correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ServiceAgentSubscriptionPayment
        billingAction={
          Object {
            "addBillingCardDetails": [Function],
            "getOrganizationBillingDetails": [Function],
            "updateBillingAddress": [Function],
            "updateCardDetails": [Function],
          }
        }
        billingDetails={Object {}}
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
      />,
      "value": null,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "billingAction": Object {
          "addBillingCardDetails": [Function],
          "getOrganizationBillingDetails": [Function],
          "updateBillingAddress": [Function],
          "updateCardDetails": [Function],
        },
        "billingDetails": Object {},
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": Object {
          "$$typeof": Symbol(react.context),
          "_calculateChangedBits": null,
          "_context": [Circular],
        },
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": null,
        "_currentValue2": null,
        "_threadCount": 0,
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ServiceAgentSubscriptionPayment
          billingAction={
            Object {
              "addBillingCardDetails": [Function],
              "getOrganizationBillingDetails": [Function],
              "updateBillingAddress": [Function],
              "updateCardDetails": [Function],
            }
          }
          billingDetails={Object {}}
          store={
            Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            }
          }
        />,
        "value": null,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "billingAction": Object {
            "addBillingCardDetails": [Function],
            "getOrganizationBillingDetails": [Function],
            "updateBillingAddress": [Function],
            "updateCardDetails": [Function],
          },
          "billingDetails": Object {},
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": Object {
            "$$typeof": Symbol(react.context),
            "_calculateChangedBits": null,
            "_context": [Circular],
          },
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": null,
          "_currentValue2": null,
          "_threadCount": 0,
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
